### DISCUSSION TITLE
Proxmox pass through network interface

### DISCUSSION THREAD

**Discussion by rafnizp** [netbox-community/netbox#18116] [General]
Hi 
I am currently working with Proxmox and NetBox, and I need some help with passing through a network interface from Proxmox to a virtual machine (VM). How do I map this in NetBox? Is there even a way to do that?

**Comment by candlerb** [comment]
Probably the best way today is:
* Create a VLAN for the subnet the VM is on (you can also link the Prefix to this VLAN)
* Configure the 802.1Q Mode of the VM NIC to "access", and select the VLAN
* Configure the 802.1Q Mode of the physical NIC to either access or trunk as appropriate, and enable the relevant VLAN(s)

Then you have an implicit link from the VM to the NIC carrying that VLAN.

Apart from this, there isn't particularly good modelling of software bridges in Netbox. You can create a virtual interface of type "bridge", and you can configure both a VM NIC and a physical NIC to have a "bridged interface" attribute which points at this bridge, but you can't directly associate a physical interface with multiple bridges, which you'd need for tagged VLANs. So you'd have to create virtual subinterfaces (like `eth0.100`, `eth0.101` etc) with the "parent" set to eth0 and the "bridge" set to the correct bridge. This makes it a bit awkward to trace through from a VM NIC to the physical interface.

**Comment by DamienUKCEH** [comment]
do you mean so-iov pass through?
